### DISCUSSION TITLE
RSU as Semaphore

### DISCUSSION THREAD

**Discussion by RandomUserM12** [eclipse/mosaic#388] [Q&A]
Is it possible for RSU to detect all the lanes on an interception and act as a Semaphore, like a light Control group, and make so the RSU sends messages to vehicles on each lane telling them when to stop (red light) and when to go (green light)?

**Comment by schwepmo** [comment]
Hey @RandomUserM12, with a bit of application logic this should be possible to achieve. If I understand you correctly, you want to detect vehicles on incoming lanes at an intersection and control a traffic light group accordingly.
You can read data from SUMO induction loops or traffic cameras using **Traffic Management Center** Application units (https://eclipse.dev/mosaic/docs/simulators/application_mapping/) and set traffic signal states using the **Traffic Lights** units.
For further reading I encourage you to check out this tutorial https://eclipse.dev/mosaic/tutorials/tiergarten_traffic_lights/ and also refer to the detector documentation of SUMO https://sumo.dlr.de/docs/Simulation/Output/#simulated_detectors

Using the resources you should be able to come up with a solution one way or the other, if you need further assistance feel free to ask.
